cro-clib/src/basic_wallet.rs

```diff
@@ -0,0 +1,160 @@
+use crate::types::{CroAddress, CroAddressPtr, CroResult};
+use chain_core::init::address::RedeemAddress;
+
+use chain_core::state::account::StakedStateAddress;
+use chain_core::tx::data::address::ExtendedAddr;
+use client_common::MultiSigAddress;
+use client_common::{PrivateKey, PublicKey};
+use std::ptr;
+
+/// # Safety
+unsafe fn do_cro_basic_create_staking_address(
+    address_out: *mut CroAddressPtr,
+    private: PrivateKey,
+) -> CroResult {
+    let public = PublicKey::from(&private);
+    let address = StakedStateAddress::BasicRedeem(RedeemAddress::from(&public));
+
+    match address {
+        StakedStateAddress::BasicRedeem(redeem) => {
+            assert!(20 == redeem.0.len());
+            let raw = redeem.to_vec();
+
+            let ret = CroAddress {
+                privatekey: private,
+                publickey: public,
+                raw,
+                address: address.to_string(),
+            };
+            let address_box = Box::new(ret);
+            ptr::write(address_out, Box::into_raw(address_box));
```

Comment:
this is the only "unsafe" part of this internal function -- perhaps the function could return `CroAddress` and the caller will then do `ptr::write` ?

Reply:
ok